### DISCUSSION TITLE
Please create an add project command

### DISCUSSION THREAD

**Discussion by louiss0** [nrwl/nx#15889] [Feature Requests]
# Nx Add Project  

There is no easy way to migrate an app from a poly to mono-repo. I decided to just create new app folders then transfer the files from my Git Hub into the app folders  myself. Ever Since I have not felt good about configuring a `project.json` file. I had create the app folders  by using  `@nxtensions/astro`. The thing is that there is no command to create a `project.json` file. I have come up with a way to do this. 

There are two aspects to this the Nx Console and the Nx Command. In a `project.json` file. the user has to have the following things. 

- A name 
- A Schema 
- Project type 
- Targets. 

**So to fulfill these conditions the user will be prompted for them in the following ways.** 

- Nx Console
- A command Prompt

## Command Prompt 

The command prompt is the prompt for how to create the `project.json` file. The user won't be asked about everything but just the things necessary to create the `project.json` file and make it usable.

**The command is called:**

```bash
nx add project 
```

This command expects a folder with a `package.json` file It the folder has one then it will then ask the user to fill in information about the file.  The user will asked the project type and targets. When asking for targets it will   keep doing so until the user says no more then the `project.json` file will be created. 

When asking for a target it will ask for a name then a command or executor. The user types in both then it will ask for another target until the user says no. If the user says no then the `project.json` will be generated.  

**When the `project.json` file is generated the following things will happen.**

- The `name` will be the name that is in the `package.json` file. 
- The `schema` is auto injected  the way all of them are. 
- The `source root` is the path that is specified 
- The **targets object** will be filled with the information from top to bottom  based on the targets requested by the prompt.   

## The Form 

The form will come from the **add project** command. It will be seen when the user clicks on a folder then clicks add. Or when the user selects Add Project UI from the Nx Console Extension Choice. 

If the user picks a folder then the form will have. The **source root** input and the targets button. 

The **source root** prompt is a prompt that will make the user fill in where the source folder is. 

The `targets ` button will make the user add a target. For each target the user would have to fill in either an `executor` or a `command` and will have to fill out the name of the target. **Remember the name has to be the name of the target inside of the targets object**. After the user is done then the project file can be created and the `project.json` file will be the same as if one that would be created if the user was prompted.

**Comment by moatorres** [comment]
@louiss0, have you tried [Nx Console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console) extension for VSCode? It has what you're looking for (and much more). It is literally a game changer. 🥇

```
Name: Nx Console
Id: nrwl.angular-console
Description: The UI for Nx & Lerna
Version: 17.32.0
Publisher: nrwl
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console
```

**Reply by louiss0** [reply]
> @louiss0, have you tried [Nx Console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console) extension for VSCode? It has what you're looking for (and much more). It is literally a game changer. 🥇
> 
> ```
> Name: Nx Console
> Id: nrwl.angular-console
> Description: The UI for Nx & Lerna
> Version: 17.32.0
> Publisher: nrwl
> VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console
> ```

I have. I know that the Nx finds all of your scripts for you but creating a `project.json` file allows the developer to integrate with Nx better. 
You must remember that an intergrated setup is not a set up where you have dependencies spread  out. In an integrated setup there aren't supposed to be any scripts.

**Comment by moatorres** [comment]
I was somehow unaware that "in an integrated setup there aren't supposed to
be any scripts". I currently manage a couple of integrated monorepos with
lint, test, build, storybook and custom scripts, such as docker and
additional build scripts. Maybe I misunderstood your request. Anyway,
hopefully other people can be aware the extension's features through this
discussion. Cheers!

**Reply by louiss0** [reply]
> I was somehow unaware that "in an integrated setup there aren't supposed to be any scripts". I currently manage a couple of integrated monorepos with lint, test, build, storybook and custom scripts, such as docker and additional build scripts. Maybe I misunderstood your request. Anyway, hopefully other people can be aware the extension's features through this discussion. Cheers!
> On Mon, 27 Mar 2023 at 13:14, shelton louis ***@***.***> wrote: @louiss0 <https://github.com/louiss0>, have you tried Nx Console <https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console> extension for VSCode? It has what you're looking for (and much more). It is literally a game changer. 🥇 Name: Nx Console Id: nrwl.angular-console Description: The UI for Nx & Lerna Version: 17.32.0 Publisher: nrwl VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console I have. I know that the Nx finds all of your scripts for you but creating a project.json file allows the developer to integrate with Nx better. You must remember that an intergrated setup is not a set up where you have dependencies spread out. In an integrated setup there aren't supposed to be any scripts. — Reply to this email directly, view it on GitHub <[#15889 (reply in thread)](https://github.com/nrwl/nx/discussions/15889#discussioncomment-5443436)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKUFGKKUKAKXMIDANZYNRU3W6G4G5ANCNFSM6AAAAAAWHFPRRU> . You are receiving this because you commented.Message ID: ***@***.***>
> -- *Moa Torres* | Moka Tecnologia
Wait a minute I can write scripts in the `package.json` file of any project and it will just work regardless of where the node modules folder is.
